docs/guide/writing-tests-for-modules.md

```diff
@@ -0,0 +1,236 @@
+# Writing tests for modules
+
+Tests are useful to guarantee that the requirements of your application are met over time.
+
+You don't have to test over and over again the same scenarios, it's a time saving tool.
+
+There are many kind of tests
+
+- [unit testing](https://en.wikipedia.org/wiki/Unit_testing) where we test a single part of the application, a single
+    module, file commonly referred as a unit
+- [integration testing](https://en.wikipedia.org/wiki/Integration_testing) where we test multiple parts of an
+    application working together as a group
+- [system testing](https://en.wikipedia.org/wiki/System_testing), sometimes referred as [end-to-end testing](https://www.indeed.com/career-advice/career-development/end-to-end-testing) where we test the entire application
+- and many more
+
+This documentation focuses on how to test Apostrophe modules, not Apostrophe itself using integration testing.
+
+Modules should be stand-alone npm packages.
+
+We will use a fictitious module named `article` to illustrate how to write tests for modules.
+
+## Requirements
+
+- A running MongoDB server
+
+## Setup
+
+This setup assumes you will use the following packages
+
+- [apostrophe](https://www.npmjs.com/package/apostrophe): *ApostropheCMS is a full-featured, open source CMS built with Node.js that seeks to empower organizations by combining in-context editing and headless architecture in a full-stack JS environment*
+- [eslint](https://www.npmjs.com/package/eslint): *ESLint is a tool for identifying and reporting on patterns found in ECMAScript/JavaScript code*
+- [eslint-config-apostrophe](https://www.npmjs.com/package/eslint-config-apostrophe): *An ESLint configuration for ApostropheCMS core and officials modules*
+- [mocha](https://www.npmjs.com/package/mocha): *Simple, flexible, fun JavaScript test framework for Node.js & The Browser*
+```sh
+npm install --save-dev apostrophe eslint eslint-config-apostrophe mocha
+```
+
+You will need to add some new scripts to your `package.json`.
+
+```json
```

Comment:
Can you wrap this in an AposCode Block with filename in the template, please. In the future we might restyle these and it will make it a lot easier.